internal/provider/provider_test.go

```
@@ -1,13 +1,110 @@
+	if meta == nil {
+		t.Fatal("Provider meta is nil")
+	}
+
+	// Test caller identity data source
+	ds := sts.DataSourceCallerIdentity()
+	if ds == nil {
+		t.Fatal("DataSourceCallerIdentity returned nil")
+	}
+
+	rd := schema.TestResourceDataRaw(t, ds.Schema, map[string]interface{}{})
+	d := ds.Read(rd, meta)
```

golangci-lint (1.62.2)

SA1019: ds.Read is deprecated: Use ReadContext or ReadWithoutTimeout instead. This implementation does not support request cancellation initiated by Terraform, such as a system or practitioner sending SIGINT (Ctrl-c). This implementation also does not support warning diagnostics. 

(staticcheck)